### PR TITLE
fix: [PIE-4819]: handle null values in execution inputs

### PR DESCRIPTION
**Pull request by vkbansal-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10564)

File: src/modules/70-pipeline/components/execution/StepDetails/tabs/ExecutionInputs/ExecutionInputs.tsx [hunk 1]
```diff
@@ -79,8 +79,12 @@ export function ExecutionInputs(props: ExecutionInputsProps): React.ReactElement
 
   const stepType = step.stepType as StepType
   const isStageForm = NonSelectableNodes.includes(step.stepType as NodeType)
-  const template = parse<{ step: StepElementConfig; stage: StageElementConfig }>(get(data, 'data.inputTemplate', '{}'))
-  const userInput = parse<{ step: StepElementConfig; stage: StageElementConfig }>(get(data, 'data.userInput', '{}'))
+  const template = parse<{ step: StepElementConfig; stage: StageElementConfig }>(
+    defaultTo(get(data, 'data.inputTemplate'), '{}')
```
Comment (udayvunnam) [inline src/modules/70-pipeline/components/execution/StepDetails/tabs/ExecutionInputs/ExecutionInputs.tsx:83]: nit: get(object: any, path: PropertyPath, defaultValue?: any) last param can be used for default value in get

**Review by dhruvUpadhyay07 — APPROVED** [review]

**Review by udayvunnam — APPROVED** [review]

File: src/modules/70-pipeline/components/execution/StepDetails/tabs/ExecutionInputs/ExecutionInputs.tsx (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/70-pipeline/components/execution/StepDetails/tabs/ExecutionInputs/ExecutionInputs.tsx:83]: If value is null, then it won't default to '{}'

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/70-pipeline/components/execution/StepDetails/tabs/ExecutionInputs/ExecutionInputs.tsx (hunk 1 above)
Comment (udayvunnam) [reply src/modules/70-pipeline/components/execution/StepDetails/tabs/ExecutionInputs/ExecutionInputs.tsx:83]: a quick check looks to be working `_.get({a: null}, a, 2)`. Am I missing something ?

Comment (vkbansal-harness) [reply src/modules/70-pipeline/components/execution/StepDetails/tabs/ExecutionInputs/ExecutionInputs.tsx:83]: `get` does not handle `null`. It only handles `undefined`. Hence this change is required

Comment (vkbansal-harness) [reply src/modules/70-pipeline/components/execution/StepDetails/tabs/ExecutionInputs/ExecutionInputs.tsx:83]: https://lodash.com/docs/4.17.15#get

**Pull request merged**